Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs

```diff
@@ -477,7 +478,15 @@ eval (RideActiveAction activeRide) state = updateAndExit state { data {activeRid
 eval RecenterButtonAction state = continue state
 
 eval (SwitchDriverStatus status) state = do
-  if state.props.driverStatusSet == status then continue state 
+  if (((getValueToLocalStore IS_DEMOMODE_ENABLED) == "true") && (status == ST.Silent || status == ST.Offline)) then do
```

Comment:
What if he is ONLINE?

Reply:
Do we need to disable the demo mode on pressing the online tab as well?